src/v2/view-builder/views/TerminalView.js

```diff
@@ -50,8 +50,10 @@ const Body = BaseForm.extend({
   showMessages () {
     const messagesObjs = this.options.appState.get('messages');
     if (messagesObjs && Array.isArray(messagesObjs.value)) {
+      // To support intentional hard refresh in case of terminal states
+      // clear sessionStorage to initiate a new login flow
+      sessionStorage.removeItem('widget-state-token');
```

Comment:
Why does this logic depend on whether there are any messages? Is there a special case of being terminal with no messages?

Reply:
as far as i remember, if the top level idx response contains `message` object then it's a terminal error. i think messagesObjs should always be non-empty in terminal error case.

Reply:
For eg this one.. It's a terminal error and only choice now user has is it to refresh.. @haishengwu-okta also this errors shows up from FormController.js at line 175
![Screen Shot 2021-01-11 at 11 30 16 AM](https://user-images.githubusercontent.com/38230587/104229018-869b7400-5400-11eb-9cc8-7832dd4ad663.png)